### PR TITLE
fix(ARQ-2169): removes exported directory if exists

### PR DESCRIPTION
**Pull request by dipak-pawar**
#### Short description of what this resolves:
Removes exported/exploded directory if present before exporting directory.

#### Changes proposed in this pull request:

- Removes directory if present
- Add unit test.

**Fixes**: #151 (ARQ-2169)

### CODE REVIEW

**Comment by lordofthejars** [comment]
LGTM, if it works in @urbandroid case, go ahead and merge it :) Also if it is urgent or blocker we can release a 1.2.1 version ASAP.

**Comment by urbandroid** [comment]
@lordofthejars of course it is not urgent or blocker as you know i already workaround with mvn clean but with this fixes like this more people will use arquillian due to not being surprising. Intuitive interfaces always better than surprising ones i think do you agree?

File: container/impl-base/src/main/java/org/jboss/arquillian/container/impl/client/deployment/ArchiveDeploymentExporter.java [hunk 1]
```diff
@@ -89,4 +91,16 @@ private String createFileName(DeploymentDescription deployment, Archive<?> archi
...
+    private void deleteIfExists(File file) {
+        if (!file.exists()) {
+            return;
+        }
+        if (file.isDirectory()) {
+            for (File sub : file.listFiles()) {
+                deleteIfExists(sub);
+            }
+        }
+        file.delete();
```
Comment (bartoszmajsak) [inline container/impl-base/src/main/java/org/jboss/arquillian/container/impl/client/deployment/ArchiveDeploymentExporter.java]: What if `.delete()` returns `false`? Shall we handle it somehow?

**Comment by bartoszmajsak** [comment]
TODO: Jira issue

**Comment by dipak-pawar** [comment]
@bartoszmajsak https://issues.jboss.org/browse/ARQ-2169

**Pull request merged**